test/com/esotericsoftware/kryo/KryoAssert.java

```diff
@@ -26,15 +26,18 @@
  */
 public class KryoAssert {
 
+    static final double EPS_D = 1.E-14;
+    static final float EPS_F = 5.E-7f;
+
     private KryoAssert() {
 
     }
 
     public static void assertDoubleEquals(double expected, double actual) {
-        assertEquals(expected, actual, 0.0);
+        assertEquals(expected, actual, EPS_D);
     }
 
     public static void assertFloatEquals(float expected, double actual) {
-        assertEquals(expected, actual, 0.0);
+        assertEquals(expected, actual, EPS_F);
     }
```

Comment:
Do all of the double and float asserts fail in your case, or only a fraction of them? If only some asserts are affected we could add the delta only where it is needed.

Reply:
I got many failures but did not check if all such assertions were failing, or only some of them. Would you be interested in the complete list of the failures I got? If so, I can remove my patch and run the tests again.

Reply:
One example each for a failing float and a failing double comparison would be great.

Just an idea: Would removing the delta from these two methods also resolve your problem?

Reply:
Without my patch I got

[ERROR] Failures: 
[ERROR]   InputOutputTest.testDoubles:742->runDoubleTest:791 positive delta expected but was: <0.0>
[ERROR]   InputOutputTest.testFloats:649->runFloatTest:697 positive delta expected but was: <0.0>
[ERROR]   UnsafeByteBufferInputOutputTest.testDoubles:700->runDoubleTest:750 positive delta expected but was: <0.0>
[ERROR]   UnsafeByteBufferInputOutputTest.testFloats:606->runFloatTest:655 positive delta expected but was: <0.0>
[ERROR]   UnsafeInputOutputTest.testDoubles:666->runDoubleTest:716 positive delta expected but was: <0.0>
[ERROR]   UnsafeInputOutputTest.testFloats:572->runFloatTest:621 positive delta expected but was: <0.0>
[ERROR]   CompatibleFieldSerializerTest.testChangePrimitiveAndWrapperFieldTypes:248->testChangePrimitiveAndWrapperFieldTypes:267 positive delta expected but was: <0.0>

There are 3 failures related to flats, 3 related to doubles, and the last one is the one concerning longs.

About your suggestion: I see in the doc of JUnit4 that assertEquals(double, double) has been deprecated and that one should ``Use assertEquals(double expected, double actual, double delta) instead''. So maybe we should stick to the positive delta fix.